src/rolling/directory.rs

```diff
@@ -210,6 +242,8 @@ pub struct RollingWriter {
     offset: usize,
     file_number: FileNumber,
     pub(crate) directory: Directory,
+    // we don't want this to be a FileNumber because it would keep that file alive
+    last_fsync: u64,
```

Comment:
can we rename this `last_fsync_file_number`. a new reader could think this is a timestamp for instance.